feat: Add spacing feature to flex example ✨

kdheepak opened the pull request:
This adds the `spacing` using `+` and `-` to the flex example:

https://github.com/ratatui-org/ratatui/assets/1813121/1980206d-8558-47f0-8631-c74ba98b224d

This was useful to find and debug spacing related bugs like this: https://github.com/ratatui-org/ratatui/pull/829

Valentin271 commented:
I like this a lot, could we display the current spacing value somewhere? And maybe add a note that `Space*` don't support spacing.

I see you fixed the proportional bug in https://github.com/ratatui-org/ratatui/pull/829 (I rebased locally and tested). I still can see a bug with `Percentage` or maybe `Length`.

https://github.com/ratatui-org/ratatui/assets/36198422/90d58a36-c8d7-4130-bf52-4f26cbdd66fe

I also recommend bumping the layout cache to 200 because every `+` or `-` is a lot of "layouting" added. I can really feel it on my machine.

kdheepak commented:
> I like this a lot, could we display the current spacing value somewhere?

Good idea. I added it to the axis.

<img width="575" alt="image" src="https://github.com/ratatui-org/ratatui/assets/1813121/9ea4714f-f0ca-4721-8f55-4e88d508f115">

I also fixed the axis to show the correct width of the blocks when there is a scrollbar or not.

> And maybe add a note that Space* don't support spacing.

Done.

<img width="576" alt="image" src="https://github.com/ratatui-org/ratatui/assets/1813121/06952894-352a-4e37-96d0-ab42817a03a5">

> I still can see a bug with Percentage or maybe Length.

Yes, that can only be fixed with this PR though: https://github.com/ratatui-org/ratatui/pull/788
Trying to get other PRs through and will work on the constraint weights at the end.

> I also recommend bumping the layout cache to 200

I bumped it to a higher number because I was seeing lag too.

Thanks for the comments!

Valentin271 reviewed (approved):
Works for me then!
